### ISSUE TITLE
Investigate validation logic for the OpenLiberty CRD

### ISSUE THREAD

**Issue by kaczyns** [OpenLiberty/open-liberty-operator#7]
The OpenLiberty CRD has no validation logic.  This may become an issue when the operator is submitted to operator hub.  An example of some validation logic in a CRD:
```yaml
  validation:
   # openAPIV3Schema is the schema for validating custom objects.
   # in kube 1.14 schemas can be version specific
    openAPIV3Schema:
      properties:
        spec:
          properties:
            license:
              type: string
              pattern: '^accept$'
            readinessProbe:
              properties:
                initialDelaySeconds: 
                  type: integer
                  minimum: 5
                  maximum: 20
            queueManager:
              properties:
                dev:
                  properties:
                    adminPassword:  
                      type: string
                      pattern: '^[a-zA-Z0-9]{8,}$'
```
We should add validation logic to the OpenLiberty CRD where possible.

**Comment by ginnick** [comment]
PR: #16

**Comment by kaczyns** [comment]
Fixed with #16.
